scripts/gha/build_desktop.py

```diff
@@ -55,14 +55,31 @@ def append_line_to_file(path, line):
     with open(path, "a") as file:
       file.write("\n" + line + "\n")
 
+
 def install_x86_support_libraries():
   """Install support libraries needed to build x86 on x86_64 hosts."""
   if utils.is_linux_os():
+    packages = ('libglib2.0-dev:i386', 'libsecret-1-dev:i386')
+
+    # First check if these packages exist on the machine already
+    for package in packages:
+      devnull = open(os.devnull, "w")
+      retval = subprocess.call(["dpkg", "-s", package], stdout=devnull, stderr=subprocess.STDOUT)
+      devnull.close()
+      # if package is not installed, dpkg returns a value other than 0 and in this case,
+      # we get out of this loop and install those packages.
+      if retval != 0:
+        break
+    else:
+      # If the for loop iterated without breaking, all required packages are already installed.
+      # Nothing to do. Return from the function.
+      return
+
     utils.run_command(['apt', 'install', 'gcc-multilib', 'g++-multilib'], as_root=True)
     utils.run_command(['dpkg', '--add-architecture', 'i386'], as_root=True)
     utils.run_command(['apt', 'update'], as_root=True)
-    utils.run_command(['apt', 'install', 'libglib2.0-dev:i386'], as_root=True)
-    utils.run_command(['apt', 'install', 'libsecret-1-dev:i386'], as_root=True)
+    for package in packages:
```

Comment:
Bit of an optimization, both dpkg and apt install can take multiple package arguments, so you can get rid of BOTH for loops and just do subprocess.call(['dpkg', '-s'] + packages, ...) and utils.run_command(['apt', 'install'] + packages).

The dpkg will return 1 if any of the packages you specified are not installed, so that should work fine for your purposes.

Reply:
Also it's probably worth querying gcc-multilib in the list too, no?

Reply:
Great tip! Didn't know there was such a convenient way to find a missing package with one command. 
Made the change and am also checking for *-multilib packages now.